src/org/opendatakit/briefcase/ui/pull/PullPanel.java

```diff
@@ -57,9 +57,9 @@
   private final Analytics analytics;
   private JobsRunner pullJobRunner;
   private TerminationFuture terminationFuture;
-  private Optional<PullSource<?>> source;
+  private Optional<PullSource> source;
 
-  public PullPanel(TransferPanelForm view, TransferForms forms, BriefcasePreferences tabPreferences, BriefcasePreferences appPreferences, TerminationFuture terminationFuture, Analytics analytics) {
+  public PullPanel(TransferPanelForm<PullSource> view, TransferForms forms, BriefcasePreferences tabPreferences, BriefcasePreferences appPreferences, TerminationFuture terminationFuture, Analytics analytics) {
```

Comment:
Now the TransferPanelForm has a type parameter that specifies the kind of panel (source or target) it is. This lets methods returning the source or target be typesafe down the chain.

Reply:
I've added my previous comment into the commit message